python_modules/libraries/dagster-azure/dagster_azure/blob/blob_fake_resource.py

```diff
@@ -0,0 +1,167 @@
+"""
+TODO sd2k: implement.
+"""
+from collections import defaultdict
+from contextlib import contextmanager
+import io
+import random
+
+from azure.core.exceptions import ResourceNotFoundError
+
+from dagster.seven import mock
+
+
+def create_blob_fake_resource(account_name):
+    '''Create a mock Blob client for test.'''
+    return BlobFakeClient(account_name, 'fake-creds')
+
+
+class BlobFakeClient(object):
+    '''Stateful mock of an ADLS2 client for testing.
+
+    Wraps a ``mock.MagicMock``. Containers are implemented using an in-memory dict.
+    '''
+
+    def __init__(self, account_name, credential):
+
+        self._account_name = account_name
+        self._credential = mock.MagicMock()
+        self._credential.account_key = credential
+        self._containers = {}
+
+    @property
+    def account_name(self):
+        return self._account_name
+
+    @property
+    def credential(self):
+        return self._credential
+
+    @property
+    def containers(self):
+        return self._containers
+
+    def get_container_client(self, container):
+        return self._containers.setdefault(
+            container, BlobFakeContainerClient(self.account_name, container)
+        )
+
+    def get_blob_client(self, container, blob):
+        return self.get_container_client(container).get_blob_client(blob)
+
+
+class BlobFakeContainerClient(object):
+    '''Stateful mock of an ADLS2 container client for testing.'''
```

Comment:
Should this say "Blob" instead of "ADLS2"?

Reply:
Yep! Fixed, thanks.